### PR TITLE
CBL-6495: Fix missing roles when pushing tasks in test-fest tests

### PR DESCRIPTION
**Pull request by pasin**
* Problem: The contributor role required by the task in the tasks' sync function is assigned to the user when the list is pushed in lists' sync function. However, the list doc and its task docs are pushed independenly and there is no order guaranteed that the list will be pushed first. As a result, if the task doc is pushed first,  the task doc will be rejected by the sync function as the contributor for the user is missing.

* Instead of using sync function to assign the contributor role to the user, assign the required roles to the users using SG REST API at the beginning of the test where the roles are being created now.

* Reorder some steps in tests to be inline with the test spec.

### CODE REVIEW

**Review by bmeike — APPROVED** [review]
Quite a big change!

File: tests/test_fest.py [hunk 1]
```diff
@@ -164,6 +176,10 @@ async def test_update_task(self, cblpytest: CBLPyTest, dataset_path: Path) -> No
             DocumentEntry("_default.tasks", "db1-list1-task1")
         ])
 
+        self.mark_test_step("Start the replicators")
```
Comment (borrrden) [inline tests/test_fest.py:179]: This should be updated in the header and test spec as one of the steps.

File: tests/test_fest.py [hunk 2]
```diff
@@ -199,6 +205,12 @@ async def test_update_task(self, cblpytest: CBLPyTest, dataset_path: Path) -> No
         verify_result = await db2.verify_documents(snapshot_updater)
         assert verify_result.result is True, f"Unexpected docs in db2: {verify_result.description}"
 
+        self.mark_test_step("Snapshot db1")
```
Comment (borrrden) [inline tests/test_fest.py:208]: Also this should be updated in the test spec.

**Review by borrrden — APPROVED** [review]
Could use some cleanup with test spec vs marked test steps.  Some of the ones I marked may already be valid, but I expect to be able to find each test step from the test spec in the test output.

File: tests/test_fest.py (hunk 1 above)
Comment (pasin) [reply tests/test_fest.py:179]: I have changed this based on the test spec.

**Comment by pasin** [comment]
Agreed, I'm working on it.

**Pull request merged**